stellarphot/core.py

```diff
@@ -62,3 +70,305 @@ def __init__(self, gain=1.0 * u.electron / u.adu,
+        'ycenter' : u.pix,
+        'fwhm_x' : u.pix,
+        'fwhm_y' : u.pix,
+        'width' : u.pix,
+        'aperture' : u.pix,
+        'annulus_inner' : u.pix,
+        'annulus_outer' : u.pix,
+        'aperture_sum' : None,
+        'annulus_sum' : None,
+        'sky_per_pix_avg' : None,
+        'sky_per_pix_med' : None,
+        'sky_per_pix_std' : None,
+        'aperture_net_cnts' : None,
+        'noise' : None,
+        'exposure' : u.s,
+        'date-obs' : None,
+        'airmass' : None,
+        'passband' : None,
+        'file' : None
+    }
+
+    def __init__(self, observatory, camera, data,  passband_map=None, retain_user_computed=False):
+        # Set metavariables
+        self.observatory = observatory
+        self.camera = camera
+        self.passband_map = passband_map
+
+        # Check the time column is correct format.
+        # Correct dtype and class
+        if (data['date-obs'].dtype != object) or (data['date-obs'][0].__class__ != Time):
+            raise ValueError(f"data['date-obs'] is not column of astropy.time.Time objects.")
+        # Correct scale is set for first element
+        if (data['date-obs'][0].scale != 'utc'):
+            raise ValueError(f"data['date-obs'] astropy.time.Time must have scale='utc', not \'{data['date-obs'][0].scale}\'.")
+
+        # Check for consistency of counts-related columns
+        counts_columns = ['aperture_sum', 'annulus_sum', 'sky_per_pix_avg', 'sky_per_pix_med',
+                          'sky_per_pix_std', 'aperture_net_cnts', 'noise']
+        cnts_unit = data[counts_columns[0]].unit
+        for this_col in counts_columns[1:]:
+            if data[this_col].unit != cnts_unit:
+                raise ValueError(f"data[{this_col}] has inconsistent units with data[{counts_columns[0]}] (should be {cnts_unit} but reported as {data[this_col].unit}).")
+
+        # Convert input data to self.data object (which will also automatically check
+        # for required columns)
+        super().__init__(self.phot_descript, data=data.copy())
+
+        # Check that self.data exists now
+        if self.data is None:
+            raise ValueError(f"self.data should exist at this point but doesn't, crap.")
+
+        # Compute additional columns (not done yet)
+        computed_columns = ['aperture_area', 'annulus_area', 'snr', 'bjd', 'night',
+                            'mag_inst', 'mag_error']
+
+        # Check if columns exist already, if they do and ignore_computed is False, throw an error
+        for this_col in computed_columns:
+            if this_col in self.data.colnames:
+                if not retain_user_computed:
+                    raise ValueError(f"Computed column '{this_col}' already exist in data.  If you want to keep them, pass retain_user_computed=True to the initializer.")
+            else:
+                # Compute the columns that need to be computed (switch requries python 3.10)
+                match this_col:
```

Comment:
I'm fine using `match` but IIRC there is some "gotcha" involved in the python implementation of there is that can make it tricky. In some subtle way that escapes me at the moment it differs from a switch in C.

Reply:
Well, thus far, this one seems to be working.